app/controllers/puppetclasses_controller.rb

```diff
@@ -9,9 +9,7 @@ class PuppetclassesController < ApplicationController
   before_action :setup_search_options, :only => :index
 
   def index
-    eager_load_tables = [:config_group_classes, :class_params, :environments,
-                         :hostgroups]
```

Comment:
After some quick tests I can see no performance difference when including `:hostgroups`. For `:environmens` there is also no impact. 

I have given this some additional thought. The view displays 20 records by default (can this even be changed?), so 90% of (eager) loaded data is thrown away in any case, since the generated query does not include a `LIMIT` clause. I really think any eager loading here is not necessary. I'm always thinking of the old proverb "premature optimization is the root of all evil". So I would argue that we need to show a case where eager loading noticeably increases performance.

But if you say so, i will re-add the loading for `:hostgrous`. :)